proc/threads_windows.go

```diff
@@ -0,0 +1,142 @@
+package proc
+
+// #include "threads_windows.h"
+import "C"
+import (
+	"fmt"
+	"unsafe"
+	"syscall"
+	sys "golang.org/x/sys/windows"
+)
+
+type WaitStatus sys.WaitStatus
+
+type OSSpecificDetails struct {
+	hThread syscall.Handle
+}
+
+func (t *Thread) halt() (err error) {
+	// TODO: We are currently ignoring the request to halt.
+	// I believe this is guaranteed to already be halted. 
+	return nil
+}
+
+func (t *Thread) singleStep() error {	
+	var context C.CONTEXT
+	context.ContextFlags = C.CONTEXT_ALL;
+	
+	res, err := C.GetThreadContext(C.HANDLE(unsafe.Pointer(t.os.hThread)), &context)
+	if res == 0 {
+		return err
+	}
+	
+	//TODO: Is it really okay to not decrement to IP?
+	//context.Rip--;
+	context.EFlags |= 0x100;
+	
+	
+	res, err = C.SetThreadContext(C.HANDLE(unsafe.Pointer(t.os.hThread)), &context)
+	if res == 0 {
+		return err
+	}
+	
+	// TODO: It's possible that this resume/trapWait will 
+	// land us on another thread. We need to guard for that 
+	// and loop here till we sucessfully land back on our thread.
+	err = t.resume()
+	if err != nil {
+		return err
+	}
+	_, err = t.dbp.trapWait(0)
+	if err != nil {
+		return err
+	}
+	
+	res, err = C.GetThreadContext(C.HANDLE(unsafe.Pointer(t.os.hThread)), &context)
+	if res == 0 {
+		return err
+	}
+		
+	context.EFlags ^= 0x100;
+	
+	res, err = C.SetThreadContext(C.HANDLE(unsafe.Pointer(t.os.hThread)), &context)
+	if res == 0 {
+		return err
+	}
+	
+	return nil
+}
+
+func (t *Thread) resume() error {
+	t.running = true
+	var res C.WINBOOL
+	t.dbp.execPtraceFunc(func() {
+		//TODO: Note that we are ignoring the thread we were asked to continue and are continuing the 
```

Comment:
One other note is that the Windows debugging APIs only allow continuing the same thread on which the previous debug event was raised.  When the debug event is raised all threads are paused and then when ContiuneDebugEvent is called all threads resume.

To correctly singleStep (or continue only a single thread while keeping others stopped), we will need to pause all the threads other than the one we want to step, then continue, then resume all the threads again after the break event is raised for the step.